Packages/com.unity.multiplayer.samples.coop/Utilities/AdditiveSceneLoader.cs

```diff
@@ -0,0 +1,99 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using Unity.Netcode;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+/// <summary>
+/// This NetworkBehavior, when added to a gameobject containing a collider (or multiple colliders) with the IsTrigger
+/// property On, loads a scene additively when there is at least one gameobject with the "Player" tag that enters its
+/// collider, and unloads it when all players leave the collider, after a specified cooldown to prevent it from
+/// repeatedly loading and unloading the same scene.
+/// </summary>
+public class AdditiveSceneLoader : NetworkBehaviour
+{
+    const float k_CooldownDuration = 5.0f;
+
+    [SerializeField]
+    string sceneName;
+
+    List<ulong> m_PlayersInTrigger;
+    bool m_IsLoaded;
+    bool m_IsCooldown;
+
+    public override void OnNetworkSpawn()
+    {
+        if (IsServer)
+        {
+            NetworkManager.OnClientDisconnectCallback += RemovePlayer;
+            m_PlayersInTrigger = new List<ulong>();
+        }
+        else
+        {
+            enabled = false;
+        }
+
+    }
+
+    public override void OnNetworkDespawn()
+    {
+        if (IsServer)
+        {
+            NetworkManager.OnClientDisconnectCallback -= RemovePlayer;
+        }
+    }
+
+    void Update()
+    {
+        if (!IsSpawned)
+        {
+            return;
+        }
+
+        if (!m_IsCooldown)
+        {
+            if (m_IsLoaded && m_PlayersInTrigger.Count == 0)
+            {
+                NetworkManager.SceneManager.UnloadScene(SceneManager.GetSceneByName(sceneName));
+                m_IsLoaded = false;
+            }
+            else if (!m_IsLoaded && m_PlayersInTrigger.Count > 0)
+            {
+                NetworkManager.SceneManager.LoadScene(sceneName, LoadSceneMode.Additive);
+                m_IsLoaded = true;
+                // Add this delay to prevent players entering and leaving the collider repeatedly from continually load/unloading the scene
+                StartCoroutine(Cooldown());
+            }
+        }
+
+    }
+
+    void OnTriggerEnter(Collider other)
+    {
+        if (other.CompareTag("Player") && other.TryGetComponent(out NetworkObject no))
+        {
+            m_PlayersInTrigger.Add(no.OwnerClientId);
+        }
+    }
+
+    void OnTriggerExit(Collider other)
+    {
+        if (other.CompareTag("Player")  && other.TryGetComponent(out NetworkObject no))
```

Comment:
If this is meant to go inside utilites, I'd create a field for the layer which triggers the scene loading/unloading, since we're hardcoding it to the Player layer at the moment.

Nitpicking: can remove the extra space, and `no`-> `networkObject`?

Reply:
>     * `ServerCharacterMovement` is throwing errors on its OnValidate method when inspecting these 3 new scenes, requiring a NavigationSystem GameObject to be present for Imps/Boss. Since we're throwing these Imps/Boss to these additive scenes, we can remove that requirement on OnValidate.

Should we replace it with a different validation?